modules/runner/src/main/ts/reporter/Reporter.ts

```diff
@@ -139,19 +179,51 @@ export const Reporter = (params: UrlParams, callbacks: Callbacks, ui: ReporterUi
     };
   };
 
+  const waitForResults = async (): Promise<void> => {
+    sendCurrentResults();
+    if (requestsInFlight.length > 0) {
+      const currentRequests = requestsInFlight.slice(0);
+      requestsInFlight.length = 0;
+      return Promise.all(currentRequests).then(() => {
+        // if more things have been queued, such as a failing test stack trace, wait for those as well
+        waitForResults();
```

Comment:
Should we be returning the promise from `waitForResults`? Although, I think since the function has the `async` identifier, JavaScript must handle it automagically.